fastapi/models/fastapi_endpoint.py

```diff
@@ -211,8 +211,24 @@ def _reset_app_cache_marker(self):
         """
 
     @api.model
-    def get_app(self, root_path):
-        record = self.search([("root_path", "=", root_path)])
+    @tools.ormcache("path")
+    def get_endpoint(self, path):
+        # try to match the request url with the most similar endpoint
+        endpoints_by_length = self.search([]).sorted(
+            lambda fe: len(fe.root_path), reverse=True
+        )
+        endpoint = False
+        while endpoints_by_length:
+            candidate_endpoint = endpoints_by_length[0]
+            if path.startswith(candidate_endpoint.root_path):
+                endpoint = candidate_endpoint
+                break
+            endpoints_by_length -= candidate_endpoint
+        return endpoint
```

Comment:
This method should handle the path as a path :smirk:. The use of the `startswith`method doesn't mean that your endpoint path is a sub url of the given path...  e.g. /a/b is not prefix of /a/bc.

see https://github.com/PyTech-SRL/rest-framework/pull/1